debezium-scripting/debezium-scripting/src/main/java/io/debezium/transforms/scripting/wasm/ChicoryEngine.java

```diff
@@ -0,0 +1,249 @@
+        var imports = ImportValues.builder()
+                .addMemory(new ImportMemory("env", "memory",
+                        new Memory(new MemoryLimits(2, memoryMax))))
+                .addFunction(ChicoryEngine_ModuleFactory.toHostFunctions(this))
+                .build();
+
+        var instanceBuilder = Instance.builder(module)
+                .withImportValues(imports);
+
+        if (aot) {
+            try {
+                instance = instanceBuilder
+                        .withMachineFactory(AotMachine::new)
+                        .build();
+            }
+            catch (ChicoryException ex) {
+                throw new DebeziumException("Failed to compile the WASM module to Java Bytecode, please use the fallback 'wasm.chicory-interpreter' ", ex);
+            }
+        }
+        else {
+            instance = instanceBuilder.build();
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private boolean aot = true;
+        private WasmModule module;
+        private int memoryMaxLimit = -1;
+
+        private Builder() {
+        };
+
+        public Builder withWasmModule(WasmModule module) {
+            this.module = module;
+            return this;
+        }
+
+        public Builder withMaxMemory(int memoryMax) {
+            this.memoryMaxLimit = memoryMax;
+            return this;
+        }
+
+        public Builder withAot(boolean aot) {
+            this.aot = aot;
+            return this;
+        }
+
+        public ChicoryEngine build() {
+            Objects.requireNonNull(module);
+            if (memoryMaxLimit > MemoryLimits.MAX_PAGES) {
+                throw new DebeziumException("Memory max limit cannot exceed: " + MemoryLimits.MAX_PAGES + " but found: " + memoryMaxLimit);
+            }
+            else if (memoryMaxLimit == -1) {
+                memoryMaxLimit = MemoryLimits.MAX_PAGES;
+            }
+            return new ChicoryEngine(aot, module, memoryMaxLimit);
+        }
+    }
+
+    private int malloc(int size) {
+        return (int) instance.export("malloc").apply(size)[0];
+    }
+
+    private void free(int ptr) {
+        instance.export("free").apply(ptr);
+    }
+
+    public Object eval(Object proxyObject) {
+        registerProxyObject(proxyObject);
+        var result = (int) instance.export("process").apply(proxyObject.hashCode())[0];
+        return objects.get(result);
+    }
+
+    public int registerProxyObject(Object proxyObjext) {
+        if (proxyObjext == null) {
+            objects.put(Integer.MAX_VALUE, null);
+            return Integer.MAX_VALUE;
+        }
+        else {
+            objects.put(proxyObjext.hashCode(), proxyObjext);
+            return proxyObjext.hashCode();
+        }
+    }
+
+    public void clearReferences() {
+        objects.clear();
+    }
+
+    // TODO: verify all the types that can go forth and back
```

Comment:
This is the biggest TODO in this PR, I'd love to have real world examples of nested `ConnectRecord` or, alternatively the edge cases to be tested.